### PR TITLE
Migrate to .NET 6 - #343

### PR DESCRIPTION
**Pull request by Exadon**
Migrate to .NET 6 LTS from .NET 5 for the .NET, C# and Ink Support.

### CODE REVIEW

**Comment by FiniteReality** [comment]
It looks like 6.0.100 doesn't exist yet; the latest version is 6.0.100-rc.1; so it will likely be some time until this will get merged properly.

**Comment by RubyNova** [comment]
@FiniteReality looks like it all passed on windows here - based on the linux failure I'm guessing this is because we've yet to update our docker container on our end. Does the specified docker image we're inheriting from exist?

**Comment by FiniteReality** [comment]
> @FiniteReality looks like it all passed on windows here - based on the linux failure I'm guessing this is because we've yet to update our docker container on our end. Does the specified docker image we're inheriting from exist?

:woman_shrugging: I'll have to check; but I think it *might* be better overall if we hold off until .NET 6 is released, rather than jumping over to it in the RC phase.

**Comment by RubyNova** [comment]
So, I can see why you'd say that - however, if the CMake is already accepting the release version number (because CMake is *weird*) then I say we go with it, as ultimately we were on .NET 5 preview before we were on stable and it was fine, I think an RC is a safer bet than a preview vesion, and I'd rather not have this PR just lingering around assuming the dockerfile is valid.

File: README.md
Comment (RubyNova) [inline README.md:35]: I think we should change this to the RC 1 version as its the readme - I don't think 6.0.100 explicitly exists yet (even though CMake is like "yeah k I'll still work")

**Review by RubyNova — APPROVED** [review]
LGTM bar one minor thing that we can change or let through but its up for discussion. :)

**Pull request merged**
